docs/kibana.asciidoc

```diff
@@ -0,0 +1,239 @@
+kind: Kibana
+metadata:
+  name: kibana-sample
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: "elasticsearch-sample"
+  config:
+     elasticsearch.requestHeadersWhitelist:
+     - authorization
+----
+
+NOTE: The configuration items you provide always take precedence over the ones that are generated by the operator.
+
+[float]
+[id="{p}-kibana-scaling"]
+=== Scaling out a Kibana deployment
+
+You may want to deploy more than one instance of Kibana. In this case all the instances have to share
+the same encryption key.
+
+This can be done either directly in the `spec.config` section:
+
+[source,yaml]
+----
+spec:
+  config:
+     xpack.security.encryptionKey: my-secret-key
+----
+
+Or by using a secure setting (recommended) as described in the next section.
+
+[float]
+[id="{p}-kibana-secure-settings"]
+=== Secure Settings
+
+You can use Kubernetes secrets to hold your secure settings in the link:k8s-es-secure-settings.html[same way it is done with Elasticsearch].
+
+As an example, if you want to scale out your Kibana deployment by creating several instances then you have to
+set `xpack.security.encryptionKey` in the Kibana configuration.
+
+. First create a secret with the secret setting:
++
+[source,yaml,subs="attributes"]
+----
+kubectl create secret generic kibana-secret-settings \
+ --from-literal=xpack.security.encryptionKey=94d2263b1ead716ae228277049f19975aff864fb4fcfe419c95123c1e90938cd
+----
++
+. Then add a reference to this secret in the `secureSettings` section:
++
+[source,yaml,subs="attributes"]
+----
+apiVersion: kibana.k8s.elastic.co/{eck_crd_version}
+kind: Kibana
+metadata:
+  name: kibana-sample
+spec:
+  version: {version}
+  count: 3
+  elasticsearchRef:
+    name: "elasticsearch-sample"
+  secureSettings:
+  - secretName: kibana-secret-settings
+----
+
+[float]
+[id="{p}-kibana-http-configuration"]
+=== HTTP Configuration
+
+[float]
+[id="{p}-kibana-http-publish"]
+==== Load balancer settings and TLS SANs
+
+By default a `ClusterIP` link:https://kubernetes.io/docs/concepts/services-networking/service/[service] is created and associated to the Kibana deployment.
+You may want to expose Kibana onto an external IP address with a link:https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer[LoadBalancer]. In this case you may also want to include a custom
+DNS name or IP in the self-generated certificate.
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: kibana.k8s.elastic.co/{eck_crd_version}
+kind: Kibana
+metadata:
+  name: kibana-sample
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: "elasticsearch-sample"
+  http:
+    service:
+      spec:
+        type: LoadBalancer # default is ClusterIP
+    tls:
+      selfSignedCertificate:
+        subjectAltNames:
+        - ip: 1.2.3.4
+        - dns: kibana.example.com
+----
+
+You can also use these two features separately, either expose the service or add some subjects to the
+self-signed certificate.
```

Comment:
I don't think these two lines are necessary.

Reply:
Same thing here with the new line. For what it's worth I am okay with the extra verbosity but understand Charith's concern since we already had described them separately above.